tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -31,6 +33,77 @@ namespace MongoDB.Driver.Tests
     public class AsyncCursorTests
     {
         //public methods
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Cursor_should_call_close_in_thread_safe_way(
+            [Values(false, true)] bool cappedCollection,
+            [Values(false, true)] bool async)
+        {
+            string testCollectionName = "test";
+            string testDatabaseName = "test";
+            var client = CreateClient();
+            DropCollection(client, testDatabaseName, testCollectionName);
+            var database = client.GetDatabase(testDatabaseName);
+
+            var createCollectionOptions = new CreateCollectionOptions();
+            if (cappedCollection)
+            {
+                createCollectionOptions.Capped = true;
+                createCollectionOptions.MaxSize = 1000;
+            }
+
+            database.CreateCollection(testCollectionName, createCollectionOptions);
+            var collection = database.GetCollection<BsonDocument>(testCollectionName);
+            collection.InsertOne(new BsonDocument());
+
+            var findOptions = new FindOptions()
+            {
+                BatchSize = 1
+            };
+            if (cappedCollection)
+            {
+                findOptions.CursorType = CursorType.TailableAwait;
+            }
+            var findResult = collection.Find(FilterDefinition<BsonDocument>.Empty, findOptions);
```

Comment:
I'm unsure if `findResult` is accurate as a variable name here since technically the variable contains a `IFindFluent` Perhaps `findFluent`?